back/tests/Api.hs

```diff
@@ -0,0 +1,300 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+module Api where
+
+import BasePrelude hiding (Category)
+import Data.Aeson
+import qualified Data.Text             as T
+import qualified Data.ByteString.Char8 as S8
+import qualified Data.Yaml             as Yaml
+import Network.HTTP.Simple
+import Control.Monad.Catch
+import Network.HTTP.Types.Status
+
+import Guide.Api.Types
+import Guide.Types.Core
+import Guide.Utils (Uid (..))
+import qualified Test.Hspec as H
+
+tests :: H.SpecWith ()
+tests = H.describe "api" $ do
+  H.it "fail request" $ do
+    request <- makeRequest
+      (Host "http://localhost/fail")
```

Comment:
It is kinda confusing because without looking at `makeRequest` it is hard to guess that `makeRequest` will sneakily add `:4400` to the host. It's better to just have something like `makeRequest "GET" "/fail"` and prepend `localhost:4400` in `makeRequest` itself.